--- 

# <span style="font-family: Verdana, Geneva, sans-serif; font-size: 28px; color: #66B2B2"><h1> Cervi-Insights: <h1/>
<h5> Unveiling Patterns in Cervical Health <h5/> 
</span> 

<div >
  <img src="https://res.cloudinary.com/dhditogyd/image/upload/v1726899939/cervical-cancer3_gcuz5e.jpg" width="100%" height=" 100%;"/>
</div>

---



Author: [Muhammad Faizan](https://www.linkedin.com/in/mrfaizanyousaf/)

<div >
  <img src="https://res.cloudinary.com/dk0hm8t43/image/upload/v1723543716/Faizan/cxhm4yhcpmponfjgjfct.jpg" width="20%" height=" 20%;"/>
</div>

## Muhammad Faizan

🎓 **3rd Year BS Computer Science** student at the **University of Agriculture, Faisalabad**  
💻 Enthusiast in **Data Analytics, Machine Learning, and Data Engineering**\
🧠 Interested in utilizing **AI/Machine Learning in HealthCare** 


### 🌐 Connect with Me

[Kaggle](https://www.kaggle.com/faizanyousafonly/) | [LinkedIn](https://www.linkedin.com/in/mrfaizanyousaf/) | [GitHub](https://github.com/faizan-yousaf/)  




### 💬 Contact Me
- **Email:** faizanyousaf815@gmail.com
- **WhatsApp:** [+92 306 537 5389](https://wa.me/923065375389)


🔗 **Let’s Collaborate:**  
I'm always open to queries, collaborations, and discussions. Let's build something amazing together!


## Meta-Data (About Dataset)

* `Cervical Cancer Risk Factors for Biopsy`:


This dataset, obtained from the UCI Repository, explores various **risk factors** associated with **cervical cancer** leading to **biopsy examinations**. Each year, about **11,000 new cases** of invasive cervical cancer are diagnosed in the U.S. Despite a steady decline in new cases over recent decades, cervical cancer remains a significant health issue, causing approximately **4,000 deaths** annually in the U.S. and around **300,000 worldwide**.

* `Age`:

Cervical cancer diagnoses predominantly occur in women aged **35 to 54**, with about **50%** of cases falling within this range. The **median age** for diagnosis is **48 years**. While it's rare for women under **20** to develop cervical cancer, many young women can become infected with multiple types of **human papilloma virus (HPV)**. Without regular screenings, these women may be at high risk for localized or invasive cancer later in life.

* `Socioeconomic and Ethnic Factors`:

Although cervical cancer rates have decreased among **Caucasian** and **African-American** women, African-American women still face higher prevalence and death rates, which are **twice** as high as their Caucasian counterparts. **Hispanic American** women also experience more than **double** the risk compared to Caucasian women, largely due to lower screening rates linked to socioeconomic factors. High **poverty levels**, lack of **health insurance**, limited **transportation**, and **language barriers** contribute to disparities in access to screening services.

* `High Sexual Activity`:

The primary risk factor for cervical cancer is **HPV**, primarily transmitted through **sexual activity**. Women with multiple sexual partners or who become sexually active at a young age are at increased risk. A history of **sexually transmitted infections**, such as **chlamydia**, has also been associated with higher cervical cancer risk.

* `Family History`:

Women with a **first-degree relative** (mother or sister) diagnosed with cervical cancer have a higher risk of developing the disease themselves.

* `Use of Oral Contraceptives`:

Long-term use of **oral contraceptives** (more than **5-10 years**) has been linked to an increased risk of cervical cancer, particularly due to a potential higher likelihood of HPV infection. The exact mechanisms behind this association remain unclear.

* `Having Many Children`:

Research indicates that having **multiple children** can elevate the risk of cervical cancer, especially among women infected with HPV.

* `Smoking`:

**Smoking** has been linked to an increased risk of precancerous changes in the cervix and progression to invasive cervical cancer, particularly in women with HPV infections.

* `Immunosuppression`:

Women with compromised **immune systems**, such as those with **HIV/AIDS**, are more vulnerable to acquiring HPV and may experience faster progression from cervical precancer to invasive cancer.

* `Diethylstilbestrol (DES)`:

Between **1938 and 1971**, the drug **diethylstilbestrol (DES)** was prescribed to pregnant women to prevent miscarriages. Daughters of these women face a heightened risk for cervical cancer, although DES is no longer prescribed.

---

This comprehensive overview of cervical cancer risk factors underscores the importance of **awareness**, **screening**, and **prevention strategies**. 

## Context

This dataset contains **835 records** and **36 columns**. It focuses on factors that may influence the likelihood of cervical cancer. The variables capture patient demographics, lifestyle factors, sexual health, and results from various medical tests.

### Content: 

#### Column Descriptions**

- **`Age`**:  
   The age of the patient in years (integer).
  
- **`Number of sexual partners`**:  
   Number of lifetime sexual partners (float).
  
- **`First sexual intercourse`**:  
   Age at first sexual intercourse (float).
  
- **`Num of pregnancies`**:  
   Number of pregnancies the patient has had (float).

- **`Smokes`**:  
   Binary indicator of whether the patient smokes (`0` = No, `1` = Yes).
  
- **`Smokes (years)`**:  
   Number of years the patient has been smoking (float).
  
- **`Smokes (packs/year)`**:  
   Average number of packs smoked per year (float).

- **`Hormonal Contraceptives`**:  
   Binary indicator for the use of hormonal contraceptives (`0` = No, `1` = Yes).
  
- **`Hormonal Contraceptives (years)`**:  
   Duration of hormonal contraceptive use in years (float).

- **`IUD`**:  
   Binary indicator for the use of an intrauterine device (`0` = No, `1` = Yes).
  
- **`IUD (years)`**:  
   Duration of IUD use in years (float).

- **`STDs`**:  
   Binary indicator for whether the patient has had sexually transmitted diseases (STDs) (`0` = No, `1` = Yes).

- **`STDs (number)`**:  
   Number of distinct STDs the patient has had (float).

- **STD-specific columns**:
   - **`STDs:condylomatosis`**: Presence of condylomatosis (genital warts) (`0` = No, `1` = Yes).
   - **`STDs:cervical condylomatosis`**: Cervical condylomatosis presence (`0` = No, `1` = Yes).
   - **`STDs:vaginal condylomatosis`**: Vaginal condylomatosis (`0` = No, `1` = Yes).
   - **`STDs:vulvo-perineal condylomatosis`**: Vulvo-perineal condylomatosis (`0` = No, `1` = Yes).
   - **`STDs:syphilis`**: Syphilis diagnosis (`0` = No, `1` = Yes).
   - **`STDs:pelvic inflammatory disease`**: Pelvic inflammatory disease (`0` = No, `1` = Yes).
   - **`STDs:genital herpes`**: Genital herpes (`0` = No, `1` = Yes).
   - **`STDs:molluscum contagiosum`**: Molluscum contagiosum (`0` = No, `1` = Yes).
   - **`STDs:AIDS`**: AIDS diagnosis (`0` = No, `1` = Yes).
   - **`STDs:HIV`**: HIV diagnosis (`0` = No, `1` = Yes).
   - **`STDs:Hepatitis B`**: Hepatitis B diagnosis (`0` = No, `1` = Yes).
   - **`STDs:HPV`**: Human papillomavirus (HPV) diagnosis (`0` = No, `1` = Yes).

- **`STDs: Number of diagnosis`**:  
   Total number of STD diagnoses (integer).

- **`STDs: Time since first diagnosis`**:  
   Time (in years) since the patient’s first STD diagnosis (float).

- **`STDs: Time since last diagnosis`**:  
   Time (in years) since the patient’s most recent STD diagnosis (float).

#### **Medical Diagnosis Columns**
  
- **`Dx:Cancer`**:  
   Binary indicator for a cancer diagnosis (`0` = No, `1` = Yes).

- **`Dx:CIN`**:  
   Binary indicator for Cervical Intraepithelial Neoplasia (CIN) diagnosis (`0` = No, `1` = Yes).

- **`Dx:HPV`**:  
   Binary indicator for HPV diagnosis (`0` = No, `1` = Yes).

- **`Dx`**:  
   General diagnosis of any health condition (`0` = No, `1` = Yes).

#### **Test Results**

- **`Hinselmann`**:  
   Result of the Hinselmann test for cervical cancer detection (`0` = Negative, `1` = Positive).

- **`Schiller`**:  
   Result of the Schiller test (`0` = Negative, `1` = Positive).

- **`Citology`**:  
   Result of the cytological test (`0` = Negative, `1` = Positive).

- **`Biopsy`**:  
   Result of a cervical biopsy (`0` = Negative, `1` = Positive).


This dataset captures a wide range of **risk factors for cervical cancer** and **related medical conditions**, including lifestyle factors, sexual health history, and results of diagnostic tests. It may be used for research in **predicting cervical cancer risk** based on these factors.

## Aims and Objectives:

We will fill this after doing the EDA and Data Preprocessing.

### Import Libraries:

Let's start the project by importing all the libraries that we will use in this project.

In [1]:
# import libraries:

# 1. to handel the data:
import numpy as np
import pandas as pd

# 2. to visualize the data:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# 3. to preprocess the data:
from sklearn.preprocessing import StandardScaler, MinMaxScaler, LabelEncoder, OneHotEncoder
from sklearn.impute import SimpleImputer, KNNImputer
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer

# 4. to build the model:
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV, RandomizedSearchCV

# 5. for classification task:
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor, AdaBoostClassifier, GradientBoostingClassifier
from sklearn.svm import SVC
from xgboost import XGBClassifier
from sklearn.neighbors import KNeighborsClassifier

# 6. Metrics:

from sklearn.metrics import accuracy_score, precision_score, recall_score, r2_score, f1_score , classification_report, mean_absolute_error, mean_absolute_percentage_error, mean_squared_error

# 7. to ignore the warnings:
import warnings
warnings.filterwarnings("ignore")

print("Libraries have been loaded successfully")

# 8. Display all rows and columns:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

Libraries have been loaded successfully


### Setting up the theme 🧬 


In [5]:
import matplotlib.pyplot as plt

print(plt.style.available)


['Solarize_Light2', '_classic_test_patch', '_mpl-gallery', '_mpl-gallery-nogrid', 'bmh', 'classic', 'dark_background', 'fast', 'fivethirtyeight', 'ggplot', 'grayscale', 'seaborn-v0_8', 'seaborn-v0_8-bright', 'seaborn-v0_8-colorblind', 'seaborn-v0_8-dark', 'seaborn-v0_8-dark-palette', 'seaborn-v0_8-darkgrid', 'seaborn-v0_8-deep', 'seaborn-v0_8-muted', 'seaborn-v0_8-notebook', 'seaborn-v0_8-paper', 'seaborn-v0_8-pastel', 'seaborn-v0_8-poster', 'seaborn-v0_8-talk', 'seaborn-v0_8-ticks', 'seaborn-v0_8-white', 'seaborn-v0_8-whitegrid', 'tableau-colorblind10']


In [2]:
# Set the primary theme colors
primary_color = '#008080'  # Teal
secondary_color = '#FFFFFF'  # White
accent_color = '#D3D3D3'  # Light Grey
highlight_color = '#66B2B2'  # Light Teal
complementary_color = '#FFE5B4'  # Soft Peach

# Matplotlib settings
plt.style.use('tableau-colorblind10')
plt.rcParams.update({
    'axes.facecolor': secondary_color,       # Set the axes background color to white
    'figure.facecolor': secondary_color,     # Set the figure background color to white
    'axes.edgecolor': accent_color,          # Light grey border
    'axes.labelcolor': primary_color,        # Teal for axis labels
    'xtick.color': primary_color,            # Teal for x-ticks
    'ytick.color': primary_color,            # Teal for y-ticks
    'grid.color': accent_color,              # Light grey grid lines
    'text.color': primary_color,             # Teal text color
    'font.size': 12,                         # Set a general font size
    'axes.titlesize': 16,                    # Title font size
    'axes.labelsize': 14,                    # Axis label font size
    'legend.fontsize': 12,                   # Legend font size
    'legend.frameon': True,
    'legend.edgecolor': highlight_color,     # Teal frame for legends
    'patch.edgecolor': primary_color,        # For borders around patches
})

# Seaborn settings
sns.set_palette([primary_color, highlight_color, complementary_color])
sns.set_style("whitegrid", {
    'grid.color': accent_color,
    'axes.edgecolor': accent_color,
    'xtick.color': primary_color,
    'ytick.color': primary_color,
    'text.color': primary_color,
})




### Load the dataset:

In [3]:
# 1. load the data from CSV file from the local machine:

df = pd.read_csv("cervical-cancer.csv")

In [ ]:
#2. load the data from the URL:
# url = "write the URL here"

# df = pd.read_csv(url)
# df.head()

# Exploratory Data Analysis (EDA)
 
Explore each column

In [4]:
df.head(10)

Age  Number of sexual partners  First sexual intercourse  \
0   18                        4.0                      15.0   
1   15                        1.0                      14.0   
2   34                        1.0                       NaN   
3   52                        5.0                      16.0   
4   46                        3.0                      21.0   
5   42                        3.0                      23.0   
6   51                        3.0                      17.0   
7   26                        1.0                      26.0   
8   45                        1.0                      20.0   
9   44                        3.0                      15.0   

   Num of pregnancies  Smokes  Smokes (years)  Smokes (packs/year)  \
0                 1.0     0.0        0.000000                  0.0   
1                 1.0     0.0        0.000000                  0.0   
2                 1.0     0.0        0.000000                  0.0   
3                 4.0     1.0       37.000000                 37.0   
4                 4.0     0.0        0.000000                  0.0   
5                 2.0     0.0        0.000000                  0.0   
6                 6.0     1.0       34.000000                  3.4   
7                 3.0     0.0        0.000000                  0.0   
8                 5.0     0.0        0.000000                  0.0   
9                 NaN     1.0        1.266973                  2.8   

   Hormonal Contraceptives  Hormonal Contraceptives (years)  IUD  IUD (years)  \
0                      0.0                              0.0  0.0          0.0   
1                      0.0                              0.0  0.0          0.0   
2                      0.0                              0.0  0.0          0.0   
3                      1.0                              3.0  0.0          0.0   
4                      1.0                             15.0  0.0          0.0   
5                      0.0                              0.0  0.0          0.0   
6                      0.0                              0.0  1.0          7.0   
7                      1.0                              2.0  1.0          7.0   
8                      0.0                              0.0  0.0          0.0   
9                      0.0                              0.0  NaN          NaN   

   STDs  STDs (number)  STDs:condylomatosis  STDs:cervical condylomatosis  \
0   0.0            0.0                  0.0                           0.0   
1   0.0            0.0                  0.0                           0.0   
2   0.0            0.0                  0.0                           0.0   
3   0.0            0.0                  0.0                           0.0   
4   0.0            0.0                  0.0                           0.0   
5   0.0            0.0                  0.0                           0.0   
6   0.0            0.0                  0.0                           0.0   
7   0.0            0.0                  0.0                           0.0   
8   0.0            0.0                  0.0                           0.0   
9   0.0            0.0                  0.0                           0.0   

   STDs:vaginal condylomatosis  STDs:vulvo-perineal condylomatosis  \
0                          0.0                                 0.0   
1                          0.0                                 0.0   
2                          0.0                                 0.0   
3                          0.0                                 0.0   
4                          0.0                                 0.0   
5                          0.0                                 0.0   
6                          0.0                                 0.0   
7                          0.0                                 0.0   
8                          0.0                                 0.0   
9                          0.0                                 0.0   

   STDs:syphilis  STDs:pelv

In [5]:
# list all the columns using for loop:

count = 0
for col in df.columns:
    count += 1
    print(count,':', col)

1 : Age
2 : Number of sexual partners
3 : First sexual intercourse
4 : Num of pregnancies
5 : Smokes
6 : Smokes (years)
7 : Smokes (packs/year)
8 : Hormonal Contraceptives
9 : Hormonal Contraceptives (years)
10 : IUD
11 : IUD (years)
12 : STDs
13 : STDs (number)
14 : STDs:condylomatosis
15 : STDs:cervical condylomatosis
16 : STDs:vaginal condylomatosis
17 : STDs:vulvo-perineal condylomatosis
18 : STDs:syphilis
19 : STDs:pelvic inflammatory disease
20 : STDs:genital herpes
21 : STDs:molluscum contagiosum
22 : STDs:AIDS
23 : STDs:HIV
24 : STDs:Hepatitis B
25 : STDs:HPV
26 : STDs: Number of diagnosis
27 : STDs: Time since first diagnosis
28 : STDs: Time since last diagnosis
29 : Dx:Cancer
30 : Dx:CIN
31 : Dx:HPV
32 : Dx
33 : Hinselmann
34 : Schiller
35 : Citology
36 : Biopsy


In [6]:
# 3. Display the shape of the data:

print("The shape of the data is: ", df.shape)
print("The data has {} rows and {} columns.".format(df.shape[0], df.shape[1]))


The shape of the data is:  (835, 36)
The data has 835 rows and 36 columns.


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 835 entries, 0 to 834
Data columns (total 36 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Age                                 835 non-null    int64  
 1   Number of sexual partners           810 non-null    float64
 2   First sexual intercourse            828 non-null    float64
 3   Num of pregnancies                  779 non-null    float64
 4   Smokes                              822 non-null    float64
 5   Smokes (years)                      822 non-null    float64
 6   Smokes (packs/year)                 822 non-null    float64
 7   Hormonal Contraceptives             732 non-null    float64
 8   Hormonal Contraceptives (years)     732 non-null    float64
 9   IUD                                 723 non-null    float64
 10  IUD (years)                         723 non-null    float64
 11  STDs                                735 non-n

In [8]:
# check the number of float, int, and object columns:

float_cols = df.select_dtypes(include=['float64']).columns
int_cols = df.select_dtypes(include=['int64']).columns
object_cols = df.select_dtypes(include=['object']).columns 
bool_cols = df.select_dtypes(include=['bool']).columns 

print("The number of float_columns is:", len(float_cols))
print("The number of integer_columns is:", len(int_cols))
print("The number of boolean_columns is:", len(bool_cols))
print("The number of object_cols is:", len(object_cols))


The number of float_columns is: 26
The number of integer_columns is: 10
The number of boolean_columns is: 0
The number of object_cols is: 0


In [9]:
# 4. Display the description of the data: (only numerical columns)

df.describe().T

count       mean       std   min   25%  \
Age                                 835.0  27.023952  8.482986  13.0  21.0   
Number of sexual partners           810.0   2.551852  1.676686   1.0   2.0   
First sexual intercourse            828.0  17.020531  2.817000  10.0  15.0   
Num of pregnancies                  779.0   2.304236  1.455817   0.0   1.0   
Smokes                              822.0   0.149635  0.356930   0.0   0.0   
Smokes (years)                      822.0   1.253850  4.140727   0.0   0.0   
Smokes (packs/year)                 822.0   0.465823  2.256273   0.0   0.0   
Hormonal Contraceptives             732.0   0.651639  0.476777   0.0   0.0   
Hormonal Contraceptives (years)     732.0   2.302916  3.794180   0.0   0.0   
IUD                                 723.0   0.114799  0.319000   0.0   0.0   
IUD (years)                         723.0   0.527621  1.965439   0.0   0.0   
STDs                                735.0   0.107483  0.309937   0.0   0.0   
STDs (number)                       735.0   0.180952  0.568153   0.0   0.0   
STDs:condylomatosis                 735.0   0.059864  0.237396   0.0   0.0   
STDs:cervical condylomatosis        735.0   0.000000  0.000000   0.0   0.0   
STDs:vaginal condylomatosis         735.0   0.005442  0.073620   0.0   0.0   
STDs:vulvo-perineal condylomatosis  735.0   0.058503  0.234853   0.0   0.0   
STDs:syphilis                       735.0   0.024490  0.154669   0.0   0.0   
STDs:pelvic inflammatory disease    735.0   0.001361  0.036886   0.0   0.0   
STDs:genital herpes                 735.0   0.001361  0.036886   0.0   0.0   
STDs:molluscum contagiosum          735.0   0.001361  0.036886   0.0   0.0   
STDs:AIDS                           735.0   0.000000  0.000000   0.0   0.0   
STDs:HIV                            735.0   0.024490  0.154669   0.0   0.0   
STDs:Hepatitis B                    735.0   0.001361  0.036886   0.0   0.0   
STDs:HPV                            735.0   0.002721  0.052129   0.0   0.0   
STDs: Number of diagnosis           835.0   0.089820  0.306335   0.0   0.0   
STDs: Time since first diagnosis     71.0   6.140845  5.895024   1.0   2.0   
STDs: Time since last diagnosis      71.0   5.816901  5.755271   1.0   2.0   
Dx:Cancer                           835.0   0.021557  0.145319   0.0   0.0   
Dx:CIN                              835.0   0.010778  0.103320   0.0   0.0   
Dx:HPV                              835.0   0.021557  0.145319   0.0   0.0   
Dx                                  835.0   0.028743  0.167182   0.0   0.0   
Hinselmann                          835.0   0.041916  0.200518   0.0   0.0   
Schiller                            835.0   0.087425  0.282626   0.0   0.0   
Citology                            835.0   0.051497  0.221142   0.0   0.0   
Biopsy                              835.0   0.064671  0.246091   0.0   0.0   

                                     50%   75%   max  
Age                                 26.0  32.0  84.0  
Number of sexual partners            2.0   3.0  28.0  
First sexual intercourse            17.0  18.0  32.0  
Num of pregnancies                   2.0   3.0  11.0  
Smokes                               0.0   0.0   1.0  
Smokes (years)                       0.0   0.0  37.0  
Smokes (packs/year)                  0.0   0.0  37.0  
Hormonal Contraceptives              1.0   1.0   1.0  
Hormonal Contraceptives (years)      0.5   3.0  30.0  
IUD                                  0.0   0.0   1.0  
IUD (years)                          0.0   0.0  19.0  
STDs                                 0.0   0.0   1.0  
STDs (number)                        0.0   0.0   4.0  
STDs:condylomatosis                  0.0   0.0   1.0  
STDs:cervical condylomatosis         0.0   0.0   0.0  
STDs:vaginal condylomatosis          0.0   0.0   1.0  
STDs:vulvo-perineal condylomatosis   0.0   0.0   1.0  
STDs:syphilis                        0.0   0.0   1.0  
STDs:pelvic inflammatory disease     0.0   0.0   1.0  
STDs:genital herpes                  0.0   0.0   

In [11]:
# 5. Display the missing values in the data:

missing_values = df.isnull().sum().sort_values(ascending=False)
missing_values


STDs: Time since first diagnosis      764
STDs: Time since last diagnosis       764
IUD                                   112
IUD (years)                           112
Hormonal Contraceptives               103
Hormonal Contraceptives (years)       103
STDs:HPV                              100
STDs:AIDS                             100
STDs:Hepatitis B                      100
STDs:HIV                              100
STDs                                  100
STDs:cervical condylomatosis          100
STDs:vulvo-perineal condylomatosis    100
STDs:syphilis                         100
STDs:pelvic inflammatory disease      100
STDs:vaginal condylomatosis           100
STDs:genital herpes                   100
STDs:molluscum contagiosum            100
STDs:condylomatosis                   100
STDs (number)                         100
Num of pregnancies                     56
Number of sexual partners              25
Smokes                                 13
Smokes (packs/year)               

# Observations:
1. The dataset has 836 rows and 36 columns.
2. The dataset has `26 float` columns, `10 integer` columns. 
3. The dataset has `0 boolean` columns, `0 object` columns
4. There are missing values in `26 columns`.
5. There are 2 columns which has the maximum number of missing values.
6. The 1st column having maximum missing values is `STDs: Time since first diagnosis` column.
   - count: `764`
   - percentage: `91.50`
7. The 2nd column having maximum missing values is `STDs: Time since last diagnosis` column.
   - count: `764`
   - percentage: `91.50`
8. The remaining columns have missing values in the range of `0.84% to 13.41%`.

In [17]:
# display the percentage of the missing values in the dataset:

missing_values_percentage = ((df.isnull().sum() / df.shape[0]) * 100).round(2).sort_values(ascending=False)
missing_values_percentage


STDs: Time since first diagnosis      91.50
STDs: Time since last diagnosis       91.50
IUD                                   13.41
IUD (years)                           13.41
Hormonal Contraceptives               12.34
Hormonal Contraceptives (years)       12.34
STDs:HPV                              11.98
STDs:AIDS                             11.98
STDs:Hepatitis B                      11.98
STDs:HIV                              11.98
STDs                                  11.98
STDs:cervical condylomatosis          11.98
STDs:vulvo-perineal condylomatosis    11.98
STDs:syphilis                         11.98
STDs:pelvic inflammatory disease      11.98
STDs:vaginal condylomatosis           11.98
STDs:genital herpes                   11.98
STDs:molluscum contagiosum            11.98
STDs:condylomatosis                   11.98
STDs (number)                         11.98
Num of pregnancies                     6.71
Number of sexual partners              2.99
Smokes                          

Here are the `top 10 strongest correlations` between variables in the dataset:


1. **STDs:vulvo-perineal condylomatosis** and **STDs:condylomatosis**: 0.988  
2. **STDs: Time since first diagnosis** and **STDs: Time since last diagnosis**: 0.931  
3. **STDs** and **STDs (number)**: 0.918  
4. **STDs: Number of diagnosis** and **STDs**: 0.906  
5. **STDs:condylomatosis** and **STDs (number)**: 0.899  
6. **STDs (number)** and **STDs: Number of diagnosis**: 0.897  
7. **STDs (number)** and **STDs:vulvo-perineal condylomatosis**: 0.891  
8. **Dx:Cancer** and **Dx:HPV**: 0.886  
9. **IUD (years)** and **IUD**: 0.746  
10. **Schiller** and **Biopsy**: 0.729


These high correlations suggest relationships between sexually transmitted diseases (STDs), diagnosis indicators, and certain test results related to cervical health.